File: golang/1773. Count Items Matching a Rule .go
func countMatches(items [][]string, ruleKey string, ruleValue string) int {
    result := 0
    for _, e := range items {
        if ruleKey == "type" && e[0] == ruleValue {
                result++
        } else if ruleKey == "color" && e[1] == ruleValue {
                result++
        } else if ruleKey == "name" && e[2] == ruleValue {
                result++
        }
    }
    return result
}

File: java/1773. Count Items Matching a Rule .java
// Joseph Igama
// Solution for problem #1773

import java.util.List;

public class Solution_1773
{
    class Solution {
        public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
            int matches = 0;

            for (int i = 0; i < items.size(); i++)
            {
                if(ruleKey.equals("type") && ruleValue.equals(items.get(i).get(0)))
                {
                    matches++;
                }

                if(ruleKey.equals("color") && ruleValue.equals(items.get(i).get(1)))
                {
                    matches++;
                }

                if(ruleKey.equals("name") && ruleValue.equals(items.get(i).get(2)))
                {
                    matches++;
                }
            }

            return matches;
        }
    }
}
